addons/event/report/report_event_registration.py

```diff
@@ -27,34 +27,37 @@ class ReportEventRegistration(models.Model):
     name_registration = fields.Char('Participant / Contact Name', readonly=True)
     company_id = fields.Many2one('res.company', 'Company', readonly=True)
 
-    @api.model_cr
-    def init(self):
-        """Initialize the sql view for the event registration """
-        tools.drop_view_if_exists(self.env.cr, 'report_event_registration')
+    # TOFIX this request won't select events that have no registration
+    def _select(self):
+        select_str = """SELECT
+            e.id::varchar || '/' || coalesce(r.id::varchar,'') AS id,
+            e.id AS event_id,
+            e.user_id AS user_id,
+            r.name AS name_registration,
+            r.create_date AS create_date,
+            e.company_id AS company_id,
+            e.date_begin AS event_date,
+            count(r.id) AS nbevent,
+            count(r.event_id) AS nbregistration,
+            CASE WHEN r.state IN ('draft') THEN count(r.event_id) ELSE 0 END AS draft_state,
+            CASE WHEN r.state IN ('open','done') THEN count(r.event_id) ELSE 0 END AS confirm_state,
+            CASE WHEN r.state IN ('cancel') THEN count(r.event_id) ELSE 0 END AS cancel_state,
+            e.event_type_id AS event_type_id,
+            e.seats_max AS seats_max,
+            e.state AS event_state,
+            r.state AS registration_state
+        """
+        return select_str
 
-        # TOFIX this request won't select events that have no registration
-        self.env.cr.execute(""" CREATE VIEW report_event_registration AS (
-            SELECT
-                e.id::varchar || '/' || coalesce(r.id::varchar,'') AS id,
-                e.id AS event_id,
-                e.user_id AS user_id,
-                r.name AS name_registration,
-                r.create_date AS create_date,
-                e.company_id AS company_id,
-                e.date_begin AS event_date,
-                count(r.id) AS nbevent,
-                count(r.event_id) AS nbregistration,
-                CASE WHEN r.state IN ('draft') THEN count(r.event_id) ELSE 0 END AS draft_state,
-                CASE WHEN r.state IN ('open','done') THEN count(r.event_id) ELSE 0 END AS confirm_state,
-                CASE WHEN r.state IN ('cancel') THEN count(r.event_id) ELSE 0 END AS cancel_state,
-                e.event_type_id AS event_type_id,
-                e.seats_max AS seats_max,
-                e.state AS event_state,
-                r.state AS registration_state
+    def _from(self):
+        from_str = """
             FROM
-                event_event e
-                LEFT JOIN event_registration r ON (e.id=r.event_id)
+            event_event e LEFT JOIN event_registration r ON (e.id=r.event_id)
```

Comment:
Yeah, but making changes here the priority is to modify the minimum number of lines, and it doesn't comply with general rule applied in other reports (one line, one table)